### PR TITLE
Simple string functions

### PR DESCRIPTION
**Pull request by tristaaan**
adds: 
- toUpperCase()
- toLowerCase()
- charCodeAt()
- codePointAt()
- indexOf()
- lastIndexOf()
- substr(), substrTo()
- substring(), substringTo()

### CODE REVIEW

**Comment by davidchase** [comment]
What do you guys prefer here @1-liners/owners all inclusive functions or just use one method in terms of the next.

So in the case of `substr` you can just do `drop` and `substrTo` you can just use `take` since under the hood they use `slice` which we could add to this library to help with the other methods as well... but I am up for whatever the collective decides :smile:

File: README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 **Functional tools that couldn’t be simpler.**
 
-We’re proud to present *1-liners* – a dead simple functional utility belt. **[103 one-liner functions][docs]** (and counting). Each hand-crafted with love and attention.
+We’re proud to present *1-liners* – a dead simple functional utility belt. **[113 one-liner functions][docs]** (and counting). Each hand-crafted with love and attention.
```
Comment (hemanth) [inline README.md]: We need to automate this ;)

Comment (stoeffel) [inline README.md]: It's automated.
https://github.com/1-liners/1-liners/blob/master/package.json#L20

**Comment by tomekwi** [comment]
@davidchase I like it how we have it now:

```js
plus(2, 3);      //» 5
plus('a', 'b');  //» 'ab'
```

**Comment by davidchase** [comment]
@tomekwi yes yes, i agree with that point... what i meant or i guess what i was asking is if 1-liners needs to or wants to encompass all the possible native methods as listed above in by @tristaaan ... or just use the methods we currently have in the library to achieve the same results...

so for example we currently have a `drop` method so we can do this
```js
drop(6, 'hello world'); //=> 'world'
```

which means that something like `substr(6, 'hello world') //=> 'world` seems to be unnecessary

essentially because `drop` and `take` are defined using `slice` it makes the `substr(), substrTo(), substring(), substringTo()` methods a bit un-needed.

unless we want to encompass all methods.

i hope i made some sense.

File: README.md (hunk 1 above)
Comment (hemanth) [inline README.md]: :+1:

**Comment by tomekwi** [comment]
Yeah, that’s exactly what I meant. `+` works for numbers as well as strings. `.slice` works for strings as well as arrays.

**Comment by tristaaan** [comment]
Shall I revert the last two commits, if so perhaps make a note in the docs of _take_ and _drop_ that they're to be used in place of substr methods? 
```js
'hello'.substr(2) === drop(2, 'hello') // => 'llo'
'razzmatazz'.substring(2, 5) === drop(2, take(5, 'razzmatazz')) // => 'zzm'
```

**Comment by stoeffel** [comment]
I agree with @tomekwi. @tristaaan yop a note would be good. Thx for your contribution.

**Comment by davidchase** [comment]
@stoeffel i think i wasnt clear, i was just trying to figure out if `1-liners` needed certain methods... also why not define `const slice = (x, y, xs) => str.slice(x, y);` 

then we can do `slice(2, 5, 'razzmatazz') // => 'zzm'` :smile:

**Comment by stoeffel** [comment]
@davidchase Got you, I don't think we need to add all native functions, if the same can be accomplished with a function already there (like with `drop`). `slice` would be a nice addition, @tristaaan could just rename `substrTo` and use `.slice` internally.

**Comment by tristaaan** [comment]
Dropped the substr(ing) functions, added `slice`

**Comment by stoeffel** [comment]
IMO this is ready to :ship:.

@tristaaan do you want to join the team?

**Comment by tomekwi** [comment]
> @tristaaan do you want to join the team?

:+1: :+1:

**Comment by TristanWright** [comment]
Awesome, yes :)

**Comment by tristaaan** [comment]
Awesome, yes :)
(commented with work account earlier sorry)

**Comment by stoeffel** [comment]
Done. you are now a member. welcome :smile:

**Pull request merged**